scipy/optimize/minpack.py

```diff
@@ -452,9 +445,16 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
                 cov_x = inv(dot(transpose(R), R))
             except (LinAlgError, ValueError):
                 pass
-        return (retval[0], cov_x) + retval[1:-1] + (mesg, info)
+        return (retval[0], cov_x) + retval[1:-1] + (errors[info][0], info)
     else:
-        return (retval[0], info)
+        if info not in LEASTSQ_SUCCESS:
+            if info > 0:
```

Comment:
This should be `if info > [5, 6, 7, 8]`, as it was previously. Then the removal of the `try-except` is correct; here there's no guarantee that `info` will not be >8 it seems.

Reply:
You could use `LEASTSQ_FAILURE` for `[5, 6, 7, 8]`

Reply:
Thank you very much for taking the time to review my PR!

According to the fortran code `info` can't be >8. Therefore, I think that `info>0, info==0, info<0` checks in this PR makes the code a bit easier to understand, but I don't have a strong opinion on this. I'll introduce a new global variable `LEASTSQ_FAILURE` as you`ve suggested.